### PR TITLE
Added two new BlendModes

### PR DESCRIPTION
**Pull request by reaperrr**
And used them to improve D2k effect visuals.

Most notably this one (left is bleed, right is new BlendMode):
new-blendmode1

### CODE REVIEW

File: OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs [hunk 1]
```diff
@@ -241,6 +241,16 @@ public void SetBlendMode(BlendMode mode)
 					GL.BlendFuncSeparate(BlendingFactorSrc.DstColor, BlendingFactorDest.Zero, BlendingFactorSrc.One, BlendingFactorDest.OneMinusSrcAlpha);
 					ErrorHandler.CheckGlError();
 					break;
+				case BlendMode.AdditiveLessBright:
+					GL.Enable(EnableCap.Blend);
+					ErrorHandler.CheckGlError();
+					GL.BlendFunc(BlendingFactorSrc.OneMinusDstColor, BlendingFactorDest.DstAlpha);
```
Comment (pchote) [inline OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs]: A quick google search suggests that `GL.BlendFunc(BlendingFactorSrc.OneMinusDstColor, BlendingFactorDest.One);` is occasionally defined as "Soft additive" blending.  How about changing this to that, and then naming it `BlendMode.SoftAdditive`?

**Comment by Phrohdoh** [comment]
If possible a percentage value would be very useful, instead of LessBright and an eventual AdditiveDarkButNotBlack, etc.

File: OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs (hunk 1 above)
Comment (reaperrr) [inline OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs]: Sounds better, yes.

**Comment by reaperrr** [comment]
> If possible a percentage value would be very useful, instead of LessBright and an eventual AdditiveDarkButNotBlack, etc.

Generally speaking I agree, but I'm not exactly an expert at this, even just these two took me a while to figure out.

**Comment by x-a-n-a-x** [comment]
I don't like the AdditiveLessBright effect that much. The explosion is too faded off.
Check other RTS games, explosions are not that translucent:

http://www.moddb.com/games/dune-2000/images/harkonnen-attacking#imagebox
http://www.moddb.com/games/dune-2000/images/corrino-palace#imagebox

http://www.ti.ppmsite.com/Public/epicbattle2.png
http://static.giantbomb.com/uploads/original/0/2238/256542-ts3.jpg

http://i.testfreaks.co.uk/images/products/600x400/158/command-conquer-3-kane-s-wrath.522142.jpg

http://www.freakygaming.com/gallery/strategy_games/starcraft_2/mass_zergling_attack_explosions.jpg

**Comment by reaperrr** [comment]
@x-a-n-a-x: It's only used for the deviator gas explosion, which was that faded off in the original as well, and some (not all) missile smoke trails.

The rest works as before, the 'old' Additive mode is still used for the other explosions.

**Comment by x-a-n-a-x** [comment]
oh ok. Forget my previous comment then. :)

**Comment by reaperrr** [comment]
Got TS-style translucency working, teaser for 75% translucency:
translucency75

**Comment by Mailaender** [comment]
This adds nice polish. :+1:

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/8a73f6041c47bd9bfdfedf1cbd2badee9639e673%5E...8a73f6041c47bd9bfdfedf1cbd2badee9639e673

**Comment by Mailaender** [comment]
> > If possible a percentage value would be very useful, instead of LessBright and an eventual AdditiveDarkButNotBlack, etc.

> Generally speaking I agree, but I'm not exactly an expert at this, even just these two took me a while to figure out.

Did this for you now in https://github.com/OpenRA/OpenRA/pull/7979.

**Pull request merged**